hashintel/hash
# Configure Heap Constraints on V8

## 🌟 What is the purpose of this PR?

This PR not only set sensible default for V8's heap constraints but also allow users to set them.

V8's default heap limit may not suit us and we can't set it to a fixed value.
Some users might want to run with limited heap while big simulation running on the cloud might want a massive amount.

## ⚠️ Known issues

The engine still crashes with OOM, but on a different level.

### 📜 Does this require a change to the docs?

The cli arguments doc changed.

## 🔗 Related links

[V8 flags list with description](https://chromium.googlesource.com/v8/v8/+/f046b936f889d1084eb1d743044d94799713904d/BUILD.gn)
- [Asana task](https://app.asana.com/0/1201893074552404/1201907929549872/f) _(internal)_

## 🛡 What tests cover this?

I don't think any test exist.

## ❓ How to test this?



1. Checkout the branch
2. Using the "--v8-max-heap-constraint" cli argument, set the limit to 500MB (likely need to extend the timeout too)
3. Run a simulation that take a lot of memory 
4. Make sure that the engine does crash with a V8 dump
5. Then setting the limit to a higher value, notice that the V8 OOM doesn't happen (some other part of the engine may still crash, like shared memory OOM)
